### ISSUE TITLE
Incorrect Full Name from integration

### ISSUE THREAD

**Issue by Frisoni** [intercom/intercom-wordpress#101]
Plugin uses wp_user->display_name to fill user Name in intercom. But, display name can modified and not always is set as "John Doe". Sometimes display_name uses the first part of user email which causes to set name as john_doe_17, for example.

Suggestion is to use wordpress_user->user_firstname." ".wordpress_user->user_lastname instead os display_name.

**Comment by lkraav** [comment]
Whatever the case, result should be filterable to do whatever is needed.

As a matter of fact, the settings output array might already be filterable, so you could override. @Frisoni have you checked?

**Comment by Frisoni** [comment]
Yes, I have know set name correctly with filter, but as an easy to go plugin most customers won't be able to customize using code.
I have a lot of incorrect user names in Intercom until I discovered what was going on.

**Comment by lkraav** [comment]
You might have a point - the default should perhaps be not the Display Name.

**Comment by GRSM78** [comment]
@Frisoni Would you mind sharing the filter?
@lkraav Any timeline for a fix?

**Comment by lkraav** [comment]
It's not really up to me to decide defaults on this project.

**Comment by Frisoni** [comment]
@GRSM78 This is the filter to use, BUT it doesn`t work every time. :(
add_filter( 'intercom_settings', function( $settings ) {           
  $current_user = wp_get_current_user(); 
  if($current_user->id !== 0){
    $settings['name'] = ...; // add specific logic to specify the name
  }
  return $settings;
} );

**Comment by GRSM78** [comment]
Have you tried to include a priority? Maybe the reason why it's not working every time is that the original filter is being triggered as well (but later), overwriting yours?
